### ISSUE TITLE
Cannot get variable at the end of an headless simulation

### ISSUE THREAD

**Issue by romainreuillon** [gama-platform/gama#2137]
### Steps to reproduce
```scala
 GamaTask.withDisposable(MoleSimulationLoader.loadModel(workDir / gaml)) { model =>
        GamaTask.withDisposable(MoleSimulationLoader.newExperiment(model)) { experiment =>

          val gamaParameters = model.getExperiment(experimentName.from(context)).getParameters

          for ((p, n) <- gamaInputs) {
            val parameter = gamaParameters.get(n)
            experiment.setParameter(n, toGAMAObject(p.from(context), parameter, parameter.getType))
          }

          experiment.setup(experimentName.from(context), seed.map(context(_)).getOrElse(rng().nextInt).toDouble)

          try experiment.play(
            stopCondition.map(_.from(context)).getOrElse(null),
            maxStep.map(_.from(context)).getOrElse(-1)
          )
          catch {
            case t: Throwable ⇒
              throw new UserBadDataError(
                s"""Gama raised an exception while running the simulation:
                    |""".stripMargin + t.stackStringWithMargin
              )
          }

          def gamaOutputVariables = gamaOutputs.map {
            case (n, p) =>
              val gamaValue = experiment.evaluateExpression(n) // line 148
              Variable.unsecure(p, fromGAMAObject(gamaValue, p.`type`.manifest.runtimeClass, p, n))
          }

          val resultContext = external.fetchOutputFiles(preparedContext, external.relativeResolver(workDir))
          external.checkAndClean(this, resultContext, workDir)
          resultContext ++ gamaOutputVariables
        }
      }
```
the gama model is:

```
/**
* Name: Predator agents (parent species)
* Author:
* Description: 5th part of the tutorial : Predator Prey
* Tags: inheritance
*/

model prey_predator

global {
	int nb_preys_init <- 200;
	int nb_predators_init <- 20;
	float prey_max_energy <- 1.0;
	float prey_max_transfert <- 0.1 ;
	float prey_energy_consum <- 0.05;
	float predator_max_energy <- 1.0;
	float predator_energy_transfert <- 0.5;
	float predator_energy_consum <- 0.02;
	int nb_preys -> {length (prey)};
	int nb_predators -> {length (predator)};
	list<list<int>> testlist;
	
	init {
		create prey number: nb_preys_init ; 
		create predator number: nb_predators_init ;
		
		write "Init";
		write "nb_preys_init: " + nb_preys_init;
		write "nb_predators_init: " + nb_predators_init;
		write "seed : " + seed;
		write testlist;
 	}
	
	reflex aff {
		write "Message at cycle " + cycle ;
	}
	
	reflex stop when: cycle>100 {
		do halt();
	}
}

species generic_species {
	float size <- 1.0;
	rgb color  ;
	float max_energy;
	float max_transfert;
	float energy_consum;
	vegetation_cell myCell <- one_of (vegetation_cell) ;
	float energy <- (rnd(1000) / 1000) * max_energy  update: energy - energy_consum max: max_energy ;
	
	init {
		location <- myCell.location;
	}
		
	reflex basic_move {
		myCell <- one_of (myCell.neighbours) ;
		location <- myCell.location ;
	}
		
	reflex die when: energy <= 0 {
		do die ;
	}
	
	aspect base {
		draw circle(size) color: color ;
	}
}

species prey parent: generic_species {
	rgb color <- #blue;
	float max_energy <- prey_max_energy ;
	float max_transfert <- prey_max_transfert ;
	float energy_consum <- prey_energy_consum ;
		
	reflex eat when: myCell.food > 0 {
		float energy_transfert <- min([max_transfert, myCell.food]) ;
		myCell.food <- myCell.food - energy_transfert ;
		energy <- energy + energy_transfert ;
	}
}
	
species predator parent: generic_species {
	rgb color <- #red ;
	float max_energy <- predator_max_energy ;
	float energy_transfert <- predator_energy_transfert ;
	float energy_consum <- predator_energy_consum ;
	list<prey> reachable_preys update: prey inside (myCell);
		
	reflex eat when: ! empty(reachable_preys) {
		ask one_of (reachable_preys) {
			do die ;
		}
		energy <- energy + energy_transfert ;
	}
}
	
grid vegetation_cell width: 50 height: 50 neighbors: 4 {
	float maxFood <- 1.0 ;
	float foodProd <- (rnd(1000) / 1000) * 0.01 ;
	float food <- (rnd(1000) / 1000) max: maxFood update: food + foodProd ;
	rgb color <- rgb(int(255 * (1 - food)), 255, int(255 * (1 - food))) update: rgb(int(255 * (1 - food)), 255, int(255 *(1 - food))) ;
	list<vegetation_cell> neighbours  <- (self neighbors_at 2); 
}

experiment prey_predatorExp type: gui {
  parameter "Test List" var: testlist;
	parameter "Nb Preys: " var: nb_preys_init  min: 0 max: 1000 category: "Prey" ;
	parameter "Prey max energy: " var: prey_max_energy category: "Prey" ;
	parameter "Prey max transfert: " var: prey_max_transfert  category: "Prey" ;
	parameter "Prey energy consumption: " var: prey_energy_consum  category: "Prey" ;
	parameter "Nb predators: " var: nb_predators_init  min: 0 max: 200 category: "Predator" ;
	parameter "Predator max energy: " var: predator_max_energy category: "Predator" ;
	parameter "Predator energy transfert: " var: predator_energy_transfert  category: "Predator" ;
	parameter "Predator energy consumption: " var: predator_energy_consum  category: "Predator" ;
	
	output {
		display main_display {
			grid vegetation_cell lines: #black ;
			species prey aspect: base ;
			species predator aspect: base ;
		}
		monitor "Number of preys" value: nb_preys;
		monitor "Number of predators" value: nb_predators;
	}
}
 
```

### Expected behavior
Tell us what should happen (include all relevant expectations)

Should evaluate the gama expression correctly

### Actual behavior
Tell us what happens instead (include relevant information, like logs, snapshots, etc.)

```
Caused by: org.openmole.core.exception.UserBadDataError: Gama raised the exception:
 | The variable or species 'predator_max_energy' cannot be resolved (either it is not defined or not accessible)
 | 	at msi.gama.runtime.exceptions.GamaRuntimeException.error(GamaRuntimeException.java:67)
 | 	at msi.gama.lang.gaml.expression.GamlExpressionCompiler.getEObjectOf(GamlExpressionCompiler.java:1134)
 | 	at msi.gama.lang.gaml.expression.GamlExpressionCompiler.compile(GamlExpressionCompiler.java:177)
 | 	at msi.gaml.expressions.GamlExpressionFactory.createExpr(GamlExpressionFactory.java:137)
 | 	at msi.gama.util.GAML.compileExpression(GAML.java:370)
 | 	at msi.gama.util.GAML.compileExpression(GAML.java:361)
 | 	at msi.gama.headless.core.Experiment.compileExpression(Experiment.java:158)
 | 	at msi.gama.headless.core.Experiment.evaluateExpression(Experiment.java:166)
 | 	at org.openmole.plugin.task.gama.GamaTask$$anonfun$process$1$$anonfun$apply$5$$anonfun$apply$7$$anonfun$gamaOutputVariables$1$1.apply(GamaTask.scala:148)
```

### System and version 
Which version of GAMA, JDK, OS ... (if necessary, copy Help>Installation Details>Configuration in GAMA)

**Comment by romainreuillon** [comment]
Error still present with commit 5ec8407f0b7f892b2a0dd1205ed86c5f9a7ba33b

**Comment by romainreuillon** [comment]
This is weird but evaluating the expression before ```play``` make the evaluation after play work properly. This code works:
```
          experiment.setup(experimentName.from(context), seed.map(context(_)).getOrElse(rng().nextInt).toDouble)

          val gamaValue = experiment.evaluateExpression("1 + 1")

          //FIXME workaround some wierd gama bug, otherwise output cannot be evaluated
          gamaOutputs.foreach {
            case (n, p) => experiment.evaluateExpression(n)
          }

          try experiment.play(
            stopCondition.map(_.from(context)).getOrElse(null),
            maxStep.map(_.from(context)).getOrElse(-1)
          )
          catch {
            case t: Throwable ⇒
              throw new UserBadDataError(
                s"""Gama raised an exception while running the simulation:
                    |""".stripMargin + t.stackStringWithMargin
              )
          }
```

**Comment by gnoubi** [comment]
Hi Romain,
Does the simulation is released when you try to evaluate the expression ? In addition, to be evaluated, a time step is required 

nicolas
-- 
Nicolas Marilleau
UMI 209 UMMISCO Institut de Recherche pour le Développement
Chercheur invité DISC/ Femto-ST Université de Franche-Comté

32 rue Henri Varagnat
93143 Bondy Cedex
tel : 01.48.02.79.01
mobile : 06.88.33.49.06

**Comment by AlexisDrogoul** [comment]
From what I've seen (briefly), the simulation was *not* disposed when Romain tried to have access to its attributes. And there is no reason to have a time step for evaluating an expression. The error obtained is typical of a bad compilation/evaluation context (for example, if the Model or ModelDescription had been disposed before).

**Comment by AlexisDrogoul** [comment]
Now tracked under #2453. Check however if it is not solved by #2426.

**Comment by gnoubi** [comment]
should be tested again with the last version of Gama and OpenMole
